app/src/apkinstaller.h

```diff
@@ -0,0 +1,55 @@
+#ifndef APK_INSTALLER_H
+#define APK_INSTALLER_H
+
+#define APK_QUEUE_SIZE 16
+#define MAX_FILENAME_SIZE 1024
+
+
+#include "apkinstaller.h"
+
+#include <SDL2/SDL_mutex.h>
+#include <SDL2/SDL_stdinc.h>
+#include <SDL2/SDL_thread.h>
+#include "command.h"
+
+// NOTE(AdoPi) apk_queue and control_event can use a generic queue
+
+struct apk_queue {
+    char data[APK_QUEUE_SIZE][MAX_FILENAME_SIZE];
+    int tail;
+    int head;
+};
+
+struct installer {
+    const char* serial;
+    SDL_Thread *thread;
+    SDL_mutex *mutex;
+    SDL_cond *event_cond;
+    SDL_bool stopped;
+    SDL_bool initialized;
+    process_t current_process;
+    struct apk_queue queue;
+};
+
+#define INSTALLER_INITIALIZER {     \
+    .serial = NULL,                 \
+    .thread = NULL,                 \
+    .mutex = NULL,                  \
+    .event_cond = NULL,             \
+    .stopped = SDL_FALSE,           \
+    .initialized = SDL_FALSE,       \
+    .current_process = PROCESS_NONE \
+}
```

Comment:
To initialize an object, either we can use a static initializer or an init method, but if both are provided they should be equivalent.